app/Imports/HistoricalDues.php

```diff
@@ -0,0 +1,452 @@
+use Maatwebsite\Excel\Concerns\OnEachRow;
+use Maatwebsite\Excel\Concerns\WithHeadingRow;
+use Maatwebsite\Excel\Concerns\WithProgressBar;
+use Maatwebsite\Excel\Row;
+use OITNetworkServices\BuzzAPI;
+use OITNetworkServices\BuzzAPI\Resources;
+use Throwable;
+
+class HistoricalDues implements WithHeadingRow, WithProgressBar, OnEachRow
+{
+    use Importable;
+
+    private const USER_CREATION_REASON_STRING = 'historical_dues_import';
+
+    /**
+     * The calling command, used for interactively resolving ambiguous records.
+     *
+     * @var \App\Console\Commands\ImportHistoricalDues
+     */
+    private $command;
+
+    /**
+     * The fiscal year being imported.
+     *
+     * @var int
+     */
+    private $fiscalYear;
+
+    private const VALID_SHIRT_SIZES = [  // couldn't figure out how to add xs to enum
+        's',
+        'm',
+        'l',
+        'xl',
+        'xxl',
+        'xxxl',
+    ];
+
+    public function __construct(ImportHistoricalDues $command, int $fiscalYear)
+    {
+        $this->command = $command;
+        $this->fiscalYear = $fiscalYear;
+    }
+
+    /**
+     * Convert row from historical dues spreadsheet to User + DuesTransaction + Payment.
+     */
+    public function onRow(Row $objRow): void
+    {
+        $row = $objRow->toArray();
+
+        $packages = self::guessDuesPackages($row, $this->fiscalYear);
+
+        if (0 === count($packages)) {
+            $this->command->newLine();
+            $this->command->error('Failed to match package(s)');
+            $this->command->table(array_keys($row), [$row]);
+
+            $packagesResponse = $this->command->ask('Enter comma-separated list of package(s) or leave blank to skip');
+
+            if (null === $packagesResponse) {
+                return;
+            }
+
+            $packages = collect(explode(',', $packagesResponse))->map(
+                static function (string $packageId): DuesPackage {
+                    return DuesPackage::where('id', trim($packageId))->firstOrFail();
+                }
+            );
+        }
+
+        try {
+            $user = self::guessUser($row);
+        } catch (Throwable $e) {
+            $this->command->newLine();
+            $this->command->error($e->getMessage());
+            $this->command->table(array_keys($row), [$row]);
+
+            $gtid = $this->command->ask('Enter correct GTID for user or leave blank to skip');
+
+            if (null === $gtid) {
+                return;
+            }
+
+            $user = User::where('gtid', $gtid)->first();
+
+            if (null === $user) {
+                CreateOrUpdateUserFromBuzzAPI::dispatchNow('gtid', $gtid, self::USER_CREATION_REASON_STRING);
+                $user = User::where('gtid', $gtid)->firstOrFail();
+            }
+        }
+
+        foreach ($packages as $package) {
+            $guessShirtSize = self::guessShirtSize($row);
+            if (null !== $guessShirtSize) {
+                $user->shirt_size = $guessShirtSize;
```

Comment:
Is it more correct to leave `polo_size` blank or copy the value from `shirt_size`?

Reply:
Leave it blank